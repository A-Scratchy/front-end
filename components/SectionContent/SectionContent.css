.SectionContent {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
}

.title {

}

.columnsContainer {
  display: flex;

}
