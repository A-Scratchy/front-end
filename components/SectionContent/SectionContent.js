import React from 'react';
import PropTypes from 'prop-types';
import styles from './SectionContent.css';

SectionContent.propTypes = {
  columns: PropTypes.array.isRequired, // can be JSX, elements, or strings
  id: PropTypes.string,
  title: PropTypes.string,
};

SectionContent.defaultProps = {
  id: undefined,
  title: undefined,
};

function SectionContent({ columns, id, title }) {
  return (
    <section className={styles.SectionContent} id={id}>
      <h3 className={styles.title}>{title}</h3>

      <div className={styles.columnsContainer}>
        {/* eslint-disable-next-line react/no-array-index-key */}
        {columns.map((column, index) => React.cloneElement(column, { key: index }))}
      </div>
    </section>
  );
}

export default SectionContent;
