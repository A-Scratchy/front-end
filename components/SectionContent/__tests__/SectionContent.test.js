/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import SectionContent from '../SectionContent';

describe('SectionContent', () => {
  it('should render with required props', () => {
    createSnapshotTest(<SectionContent columns={2}>Test</SectionContent>);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <SectionContent className="test-class" title="test-title" id="test-id" columns={3}>
        Test
      </SectionContent>,
    );
  });
});
