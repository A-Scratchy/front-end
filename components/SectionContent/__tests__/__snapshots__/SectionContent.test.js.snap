// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionContent should render with many props assigned 1`] = `
<section
  className="test-class SectionContent"
  columns={3}
  id="test-id"
  title="test-title"
/>
`;

exports[`SectionContent should render with required props 1`] = `
<section
  className="SectionContent"
  columns={2}
/>
`;
